autoload/ale/util.vim

```diff
@@ -543,3 +543,142 @@ endfunction
 function! ale#util#GetBufferContents(buffer) abort
     return join(getbufline(a:buffer, 1, '$'), '\n') . '\n'
 endfunction
+
+let s:uri_handlers = {
+\   'jdt': {
+\       'OpenURILink': function('ale#util#OpenJDTLink'),
+\       'PathFromURI': function('ale#util#JDTToPath'),
+\       'PathToURI': function('ale#util#PathToJDT')
+\   }
+\}
+
+function! ale#util#GetURIHandler(uri) abort
+    for l:scheme in keys(s:uri_handlers)
+        if a:uri =~# '^'.l:scheme.'://'
+            return s:uri_handlers[scheme]
+        endif
+    endfor
+    return v:null
+endfunction
+
+function! ale#util#JDTToPath(uri) abort
```

Comment:
Put all of the JDT logic in a separate file like `path/jdt.vim` and then we'll only load that code when needed. This will save on time needed to start up ALE for people who don't write Java code.

Reply:
See commit f397c2e.